builtin/providers/cloudstack/resource_cloudstack_network.go

```diff
@@ -94,17 +122,34 @@ func resourceCloudStackNetworkCreate(d *schema.ResourceData, meta interface{}) e
 	p := cs.Network.NewCreateNetworkParams(displaytext.(string), name, networkofferingid, zoneid)
 
 	// Get the network details from the CIDR
-	m, err := parseCIDR(d.Get("cidr").(string))
+	m := make(map[string]string, 4)
+	cidr := d.Get("cidr").(string)
+	m["cidr"] = cidr
+	if startip, ok := d.GetOk("startip"); ok {
+		m["startip"] = startip.(string)
+	}
+	if endip, ok := d.GetOk("endip"); ok {
+		m["endip"] = endip.(string)
+	}
+	if gateway, ok := d.GetOk("gateway"); ok {
+		m["gateway"] = gateway.(string)
+	}
```

Comment:
I think it would be a lot cleaner if you put all this logic in the `parseCIDR` func as now you are checking if things like `startip`, `endip` and `gateway` are set both here and again in the `parseCIDR` func.

So why not change lines 96 and 97 to something like:

```go
// Get the network details from the provided config
m, err := computeIPConfig(d)
```

And the rename `parseCIDR` to `computeIPConfig` and do all checking in that one function?